parasolr/django/tests/test_django.py

```diff
@@ -111,3 +137,47 @@ class MyAliasedSolrQuerySet(AliasedSolrQuerySet):
     # alias queryset init: field list and reverse alias lookup populated
     assert mysqs.field_list
     assert mysqs.reverse_aliases
+
+
+@skipif_no_django
+@patch('parasolr.django.queryset.SolrClient')
+def test_identify_index_dependencies(mocksolrclient):
+
+    ModelIndexable.identify_index_dependencies()
+
+    # collection model should be in related object config
+    assert Collection in ModelIndexable.related
+    # save/delete handler config options saved
+    assert ModelIndexable.related[Collection] == \
+        IndexItem.index_depends_on['collections']
+    # through model added to m2m list
+    assert IndexItem.collections.through in ModelIndexable.m2m
+
+    # dependencies should be cached on thef irst run and not regenerated
+    with patch.object(ModelIndexable, '__subclasses__') as mockgetsubs:
+        ModelIndexable.identify_index_dependencies()
+        assert mockgetsubs.call_count == 0
+
+
+@skipif_no_django
+def test_get_related_model(caplog):
+    # test app.Model notation with stock django model
+    from django.contrib.auth.models import User
+    assert ModelIndexable.get_related_model(IndexItem, 'auth.User') == User
+
+    # many to many
+    assert ModelIndexable.get_related_model(IndexItem, 'collections') == \
+        Collection
+
+    # reverse many to many
+    assert ModelIndexable.get_related_model(IndexItem, 'owner_set') == \
+        Owner
+
+    # multipart path
```

Comment:
probably doesn't matter for this comment, but wonder if we should have a clearer name for this..."multi-step relationship" or something? "multipart path" to me is more about the fact that the name of it will be split, not the actual nature of the relationship